### DISCUSSION TITLE
Should register and setValue called inside useEffect?

### DISCUSSION THREAD

**Discussion by Zerebokep** [react-hook-form/react-hook-form#9264] [Q&A] [answered]
title

**Comment by leapful** [comment, marked as answer]
@Zerebokep 

No for method "register" because the method also apply prop "ref" to the input element in order to perform other behaviors like focus on error, validation on blur ...

It depends for method "setValue". If the change dependency passed to hook "useEffect" is not included any data related to field to be set by "setValue", it's fine, otherwise it might lead to infinite re-rendering.
